src/remoteConnector.ts

```diff
@@ -789,10 +805,10 @@ export default class RemoteConnector extends Disposable {
 			const usingSSHGateway = typeof sshDest !== 'string';
 
 			const connectionSuccessful = await isRemoteExtensionHostRunning();
-			const connectionInfo = this.context.globalState.get<SSHConnectionParams>(sshDestStr);
+			const connectionInfo = this.context.globalState.get<SSHConnectionParams & { isFirstConnection: boolean }>(sshDestStr);
 			if (connectionInfo) {
 				const kind = usingSSHGateway ? 'gateway' : 'local-app';
-				if (connectionSuccessful) {
+				if (connectionInfo.isFirstConnection && connectionSuccessful) {
 					this.telemetry.sendTelemetryEvent('vscode_desktop_ssh', {
```

Comment:
I thought we reverthed this code, since it cannot be reliable? and put in previous place?